scripts/tools/nrfconnect/nrfconnect_factory_data.schema

```diff
@@ -0,0 +1,99 @@
+{
+    "$id": "NrfConnect_Factory_Data_schema",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "description": "A representation of all factory data used in Matter's NrfConnect device",
+    "type": "object",
+    "required": [
+        "sn",
+        "date",
+        "hw_ver",
+        "hw_ver_str",
+        "rd_uid",
+        "dac_cert",
+        "dac_key",
+        "pai_cert",
+        "cd",
+        "spake2_it",
+        "spake2_salt",
+        "spake2_verifier"
+    ],
+    "properties": {
+        "sn": {
+            "description": "Serial number of device",
+            "type": "string",
+            "maxLength": 32
+        },
+        "date": {
```

Comment:
json shema has a dedicated type for date/time. Maybe worth checking (i.e. to get rid of regex and lengths):
https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times

Reply:
JSON-schema provides a `date` format in ISO 8601 (YYYY-MM-DD), but in specs, we have a format YYYYMMDD, so without '-'. So, I think we should be coherent with spec.